### PR TITLE
Av updates mar2020

### PR DESCRIPTION
**Pull request by NicHaley**
### Description
*Describe what has changed here.*

---
### Checklist

- [ ] The steps for distribution have been follow [here](https://www.notion.so/adasupport/Distribution-3a9dfc90c9b3449781b6f9c825f1dee8).
- [ ] A [Release Note](https://www.notion.so/adasupport/Creating-Release-Notes-36906dfa8a2b4f10a31dc23b8d46681a) has been drafted and published after merging to master.

### CODE REVIEW

File: ExampleApp/ViewController.swift [hunk 1]
```diff
@@ -11,7 +11,7 @@ import AdaEmbedFramework
 
 class ViewController: UIViewController {
     
-    var adaFramework = AdaWebHost(handle: "kiwi")
+    var adaFramework = AdaWebHost(handle: "nic", openWebLinksInSafari: false, appScheme: "adaexampleapp")
```
Comment (aaronvegh) [inline ExampleApp/ViewController.swift]: Whoops, we should probably set this back to "kiwi" for the handle!

File: EmbedFramework/AdaWebHost.swift [hunk 2]
```diff
@@ -152,9 +198,23 @@ extension AdaWebHost: WKNavigationDelegate, WKUIDelegate {
     public func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Swift.Void) {
         if navigationAction.navigationType == WKNavigationType.linkActivated {
             if let url = navigationAction.request.url {
-                let shared = UIApplication.shared
-                if shared.canOpenURL(url) {
-                    shared.open(url, options: [:], completionHandler: nil)
+                if url.scheme == appScheme {
```
Comment (NicHaley) [inline EmbedFramework/AdaWebHost.swift]: Maybe we can add a comment here that this case is for deep link support, took me a moment to figure out that `share.open` can handle deep links as well as web URLs 😬 

As an aside, do you call them deep links or univeral links? Not sure I understand the difference

File: EmbedFramework/AdaWebHost.swift [hunk 3]
```diff
@@ -200,7 +259,12 @@ extension AdaWebHost {
     }
     
     private func evalJS(_ toRun: String) {
+        guard self.webHostLoaded else {
```
Comment (NicHaley) [inline EmbedFramework/AdaWebHost.swift]: nice and clean!

File: ExampleApp/ViewController.swift (hunk 1 above)
Comment (NicHaley) [reply ExampleApp/ViewController.swift]: Doesn't matter much to me!

**Comment by NicHaley** [comment]
> Hey Nic! I just saw the one thing that we should probably fix. I can do that if you need. But otherwise, I'm hoping you've given this a spin yourself? Let me know if there's anything else you'd like!

@aaronvegh I tried it out, seems to be working well!

File: EmbedFramework/AdaWebHost.swift [hunk 4]
```diff
@@ -17,6 +18,13 @@ public class AdaWebHost: NSObject {
     public var styles = ""
     public var greeting = ""
     
+    /// Set the chat to open links in Safari
+    /// Set to false to use SFSafariViewController instead
+    public var openWebLinksInSafari = true
```
[the remark this answers is not in the document]
Comment (aaronvegh) [reply EmbedFramework/AdaWebHost.swift]: Right, done!

File: EmbedFramework/AdaWebHost.swift [hunk 5]
```diff
@@ -31,13 +39,27 @@ public class AdaWebHost: NSObject {
...
+    }
     
     /// Keep track of whether we're showing offline view
     internal var isInOfflineMode = false
     
-    public init(handle: String, cluster: String = "", language: String = "", styles: String = "", greeting: String = "", metafields: [String: String]? = [:]) {
+    /// If commands are sent prior to `embedReady`, store until it can be cleared out
+    private var pendingCommands = [String]()
+    
+    public init(handle: String, openWebLinksInSafari: Bool, appScheme: String = "", cluster: String = "", language: String = "", styles: String = "", greeting: String = "", metafields: [String: String]? = [:]) {
```
[the remark this answers is not in the document]
Comment (aaronvegh) [reply EmbedFramework/AdaWebHost.swift]: Good call! I've created an `AdaWebHostOptions` struct and given those two new values to that as properties. And updated the code on both the framework and sample app to use that instead.

File: EmbedFramework/AdaWebHost.swift (hunk 2 above)
Comment (aaronvegh) [reply EmbedFramework/AdaWebHost.swift]: Done. "Universal Links" is Apple's term for the tech that lets you tie your web site's domain to your app so it'll handle web links into the app. I think "Deep Link" is a generic term for doing that, which maybe gets used on Android too?

**Comment by aaronvegh** [comment]
All set with these change requests! Let me know what you think.

File: EmbedFramework/AdaWebHost.swift (hunk 2 above)
Comment (NicHaley) [reply EmbedFramework/AdaWebHost.swift]: Gotcha, thanks!

**Comment by NicHaley** [comment]
@aaronvegh Looks great!!

**Pull request merged**